### PR TITLE
Added some important updates to Heart Rate detection 

### PR DESCRIPTION
**Pull request by Aryan-Prakhar**
Updated the data type of avg to uint32_t so that data may not overflow returning a negative value which is undesirable and BPM to uint8_t to maintain it as integer.

### CODE REVIEW

**Comment by Aryan-Prakhar** [comment]
@lorforlinux I have uploaded HeartRateDetection_V2.0 here also.

File: software/HeartRateDetection/HeartRateDetection.ino [hunk 1]
```diff
@@ -33,15 +34,15 @@
 
 #define SAMPLE_RATE 125
 #define BAUD_RATE 115200
-#define INPUT_PIN 36
+#define INPUT_PIN A2
```
Comment (lorforlinux) [inline software/HeartRateDetection/HeartRateDetection.ino:37]: Always mention that the user has to change this parameter, I think changing `36` to `A0` here means that the code was tested on ESP32 before and now you have tested it with Arduino UNO R3.

File: software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino [hunk 2]
```diff
@@ -0,0 +1,230 @@
+// HeartRateDetection_V2.0 - BioAmp EXG Pill
+// https://github.com/upsidedownlabs/BioAmp-EXG-Pill
+// https://github.com/upsidedownlabs/Heart-BioAmp-Arduino-Firmware
+
+// Upside Down Labs invests time and resources providing this open source code,
+// please support Upside Down Labs and open-source hardware by purchasing
+// products from Upside Down Labs!
+
+// Copyright (c) 2021 - 2024 Upside Down Labs - contact@upsidedownlabs.tech
+// Copyright (c) 2021 - 2024 Aryan Prakhar - aryanprakhar1010@gmail.com
```
Comment (lorforlinux) [inline software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino:10]: Please add Dev as a contributor here as well.

File: software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+#define BAUD_RATE 115200      // Serial communication baud rate
+#define INPUT_PIN A2          // Analog  pin for ECG signal
+#define OUTPUT_PIN 13         // Digital output pin for LED
+#define DATA_LENGTH 26        // Length of data buffer for peak detection
+#define buffer_size 5         // Size of circular arrays for BPM calculation
+
+// Global variables for BPM calculation
+int data_index = 0;
+uint32_t tdiff = 0;           // Time difference between consecutive peaks
+uint8_t BPM = 0;              // Calculated Beats Per Minute
```
Comment (lorforlinux) [inline software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino:45]: Please use a standard way of choosing variable names. I can define this under docs if it's not clear to you.

File: software/OLED_BPM/OLED_BPM.ino [hunk 4]
```diff
@@ -0,0 +1,240 @@
+ /*OLED_BPM - BioAmp EXG Pill
+ https://github.com/upsidedownlabs/BioAmp-EXG-Pill
+ https://github.com/upsidedownlabs/Heart-BioAmp-Arduino-Firmware
+
+ Upside Down Labs invests time and resources providing this open source code,
+ please support Upside Down Labs and open-source hardware by purchasing
+ products from Upside Down Labs!
+
+ Copyright (c) 2021 - 2024 Upside Down Labs - contact@upsidedownlabs.tech
+ Copyright (c) 2021 - 2024 Aryan Prakhar - aryanprakhar1010@gmail.com
```
Comment (lorforlinux) [inline software/OLED_BPM/OLED_BPM.ino:10]: Add Dev here as well.

File: software/OLED_BPM/OLED_BPM.ino [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+          display.setTextSize(3);
+          display.println(BPM);
+          display.display();
+        }
+      }  
+  }
+}
+
+// Function to detect peaks in the ECG signal
+bool Getpeak(float new_sample) {
```
Comment (lorforlinux) [inline software/OLED_BPM/OLED_BPM.ino:165]: We need to provide a detailed explanation of this function.

**Review by lorforlinux — COMMENTED** [review]
@Aryan-Prakhar Please check the comments I have added.

**Comment by Aryan-Prakhar** [comment]
@lorforlinux i have checked the comments and updated them accordingly. Please review

**Comment by Aryan-Prakhar** [comment]
@lorforlinux I have made all the necessary corrections. Please review.

File: software/HeartRateDetection/HeartRateDetection.ino (hunk 1 above)
Comment (Aryan-Prakhar) [reply software/HeartRateDetection/HeartRateDetection.ino]: I have add few comments to describe that we may need to change the input pin number in the Arduino code

File: software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino (hunk 2 above)
Comment (Aryan-Prakhar) [reply software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino]: I have add Dev as a contributor for this project

File: software/OLED_BPM/OLED_BPM.ino (hunk 4 above)
Comment (Aryan-Prakhar) [reply software/OLED_BPM/OLED_BPM.ino]: I have added Dev as a contributor mentioning his email

File: software/OLED_BPM/OLED_BPM.ino (hunk 5 above)
Comment (Aryan-Prakhar) [reply software/OLED_BPM/OLED_BPM.ino:170]: I have added few comments describing the detailed explanation of all attributes included in this function. Please review the same.

**Comment by Aryan-Prakhar** [comment]
@lorforlinux I have resolved all the changes mentioned above also few changes that was asked later on. Please review

File: software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino (hunk 3 above)
Comment (Aryan-Prakhar) [reply software/HeartRateDetection_V2.0/HeartRateDetection_V2.0.ino]: I have used lower case the variable to standardized the variables used in this program

**Comment by lorforlinux** [comment]
@Aryan-Prakhar Thank you for your contribution, I am merging now.

**Review by lorforlinux — COMMENTED** [review]
I have reviewed this.

**Review by lorforlinux — APPROVED** [review]

**Pull request merged**